app/cdap/components/GridContainer/index.tsx

```diff
@@ -14,64 +14,119 @@
 
-  const handleTransformationUpload = (valueToUpdate: string, newValue) => {
-    // valueToUpdate; // 'function', 'column'
+  const [snackbarState, setSnackbar] = useSnackbar();
 
-    setTransformationPayload((prev) => ({
-      function: valueToUpdate === 'function' ? newValue : prev.function,
-      column: valueToUpdate === 'column' ? newValue : prev.column,
-    }));
+  useEffect(() => {
+    if (snackbarState.open) {
+      setTimeout(() => {
+        setSnackbar(() => ({
+          open: false,
+        }));
+      }, 5000);
+    }
+  }, [snackbarState.open]);
 
-    // check if both function and column are selected
-    if (
-      (valueToUpdate == 'function' && transformationPayload.column) ||
-      (valueToUpdate == 'column' && transformationPayload.function)
-    ) {
-      // if yes, then do applyDirective
-      const directive = getDirective(newValue.option, transformationPayload.column);
-      // addDirectives()
+  /*
+   * In this useEffect we are handling apply transformation based on conditions
+   * whether to apply transformation directly without add transformation panel open
+   * or should pass data to add transformation step panel
+   */
+  useEffect(() => {
+    // checks if both function and column are selected
+    if (selectedColumn && selectedFunction) {
+      // If both function and column are selected, then apply transformation
+      const directive = getDirective(selectedFunction.option, selectedColumn);
       execute([directive]).subscribe(
-        () => {},
-        (error) => {
-          DataPrepStore.dispatch({
-            type: DataPrepActions.setError,
-            payload: {
-              message: error.message || error.response.message,
+        () => {
+          setSelectedColumn(false);
+          setSelectedFunction({
+            option: {
+              getUsage({}) {
+                return '';
+              },
+              label: '',
+              supportedDataType: [''],
+              value: '',
             },
+            supportedType: [''],
+          });
+          setSnackbar({
+            open: true,
+            isSuccess: true,
+            message: 'Transformation Successfully Applied',
+          });
+        },
+        (error) => {
+          setSnackbar({
+            open: true,
+            isSuccess: false,
+            message: error.message || error.response.message,
           });
         }
       );
+    } else {
+      // If selectedColumn is not selected then open add transformation panel and apply transformation
     }
-    // if no then return, do nothing
-  };
+  }, [selectedFunction, selectedColumn]);
 
-  const getDirective = (functionName: string, selectedColumnName: string | boolean) => {
-    if (DATATYPE_OPTIONS.some((eachOption) => eachOption.value === functionName)) {
-      return `set-type :${selectedColumnName} ${functionName}`;
+  const handleSelectedFunctionColumnState = (valueToUpdate: string, newValue) => {
```

Comment:
use separate functions for handling column and function

Reply:
Instead of handling with seperate functions directly setting the states without using functions here in GridTable